### ISSUE TITLE
nb_execution_mode=auto should take into account cells with no expected output

### ISSUE THREAD

**Issue by bsipocz** [executablebooks/MyST-NB#416]
### Describe the bug

**context and expectation** 
When I use `nb_execution_mode=auto` I would expect that execution is skipped when all the code cells are run.

**bug**
But instead the execution is always run.

**problem**
This is because some of the cells 
 - produces no output, even though they are run. E.g. see the first code cell here, which contains imports: https://github.com/NASA-NAVO/navo-workshop/blob/main/CS_Catalog_Queries.ipynb
 - accidentally remained empty cells
 - or maybe cells with commented out code only (especially in the case of tutorial exercise notebooks)

I think none of these cases should trigger the execution.

I'm happy to propose a fix in a PR

### Reproduce the bug

Run all the cells in the notebook referenced above, save it. Then run `sphinx-build -b html -D nb_execution_mode=auto . _build/html` on it

**Comment by chrisjsewell** [comment]
Heya, yeh happy see proposals/suggestions thanks 👍 

I'd note an alternative solution though, now we have file-level configuration: https://myst-nb.readthedocs.io/en/latest/configuration.html#file-level-configuration

if you want certain notebooks executed and others not, you could set globally `nb_execution_mode=force`, then set per-file:

```json
{
  "metadata": {
    "mystnb": {
      "execution_mode": "off"
    }
  }
}
```

or visa-versa

**Comment by bsipocz** [comment]
Yes, I think that will be a good solution for some of the projects I maintain (e.g. where I definitely want to keep the outputs as they are, e.g. when they are a product of cells with a long run time), but the `auto` option promises such a nice, low maintenance middle-ground.

**Comment by fzimmermann89** [comment]
We have a similar issue: A lot of cells in our notebooks do not produce any output. 
The notebooks are automatically created and are either fresh without any outputs, or up to date.

This, we would like a "lazy" option: Instead of running if any output is missing, run if no output exists.

**Comment by davidorme** [comment]
We have a Jupyter notebook that always gets run and I think for the same reason of having expectedly 'empty' outputs. The `lazy` mode sounds like what we are after - unless cell metadata could be used to mark silent cells. That's more effort but more precise for when a notebook needs to run?
